task/maven/0.1/maven.yaml

```diff
@@ -24,6 +24,14 @@ spec:
       description: The Maven repository mirror url
       type: string
       default: ""
+    - name: SERVER_USER
+      description: The username for the server
+      type: string
+      default: ""
+    - name: SERVER_PASSWORD
+      description: The password for the server
+      type: string
+      default: ""      
```

Comment:
Hi @soharaki, can we avoid adding more parameters to the task and instead make use of `ConfigMap` if we need to mount more settings into settings.xml? [Reference](https://github.com/tektoncd/catalog/tree/master/task/maven/0.1#with-custom-m2settingsyaml)

Reply:
Hi @vinamra28 
Here are some of the ways we want to use mirror servers in maven.
1. Communicate with maven by https.
2. Use the cached library in a private network environment.
In the latter example, I supposed that `server` tag is required, along with the mirror tag for the security.

I understand what you are saying.
If we try to represent all the tags in settings.xml in maven's Task parameter, and the we have to add so MANY parameters to the settings.xml.

```xml
    <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                          https://maven.apache.org/xsd/settings-1.0.0.xsd">
      <localRepository/>
      <interactiveMode/>
      <offline/>
      <pluginGroups/>
      <servers/>
      <mirrors/>
      <proxies/>
      <profiles/>
      <activeProfiles/>
    </settings>
```
via [settings.xml](https://maven.apache.org/settings.html)